cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -2567,19 +2578,42 @@ private CloudServicePlan findPlanForService(CloudService service) {
         }
         Map<String, Object> responseMap = JsonUtil.convertJsonToMap(response);
         List<Map<String, Object>> newResources = (List<Map<String, Object>>) responseMap.get("resources");
-        if (newResources != null && newResources.size() > 0) {
+        if (newResources != null && !newResources.isEmpty()) {
             allResources.addAll(newResources);
         }
         addAllRemainingResources(responseMap, allResources);
         return allResources;
     }
 
     private void addAllRemainingResources(Map<String, Object> responseMap, List<Map<String, Object>> allResources) {
-        String nextUrl = (String) responseMap.get("next_url");
-        while (nextUrl != null && nextUrl.length() > 0) {
-            nextUrl = addPageOfResources(nextUrl, allResources);
+        String nextUrlKey = "next_url";
+        String nextUrl = getNextUrl(responseMap, nextUrlKey);
+
+        // Check for V3 response
+        if (nextUrl == null || nextUrl.isEmpty()) {
+            nextUrlKey = "next";
+            nextUrl = getV3NextUrl(responseMap);
+        }
+        while (nextUrl != null && !nextUrl.isEmpty()) {
+            nextUrl = addPageOfResources(nextUrl, allResources, nextUrlKey);
         }
     }
+    
+    private String getV3NextUrl(Map<String, Object> responseMap) {
```

Comment:
My idea was to have a method called `getNextUrl` that tries to get the next URL through the `next_url` field of the response and if that doesn't work, tries to get it from the `pagination/next` field. That way you wouldn't need to pass the `next` parameter to the `addPageOfResources` method.